### ISSUE TITLE
make harmonics optional/controllable

### ISSUE THREAD

**Issue by kyle-woodward** [kyle-woodward/kaza-lc#3]
Changes to be made in [02sentinel2_sr.py](https://github.com/kyle-woodward/kaza-lc/blob/main/02sentinel2_sr.py).
Currently harmonics (phase & amplitude) are computed on the whole year's worth of S2 images, applied to the NIR and SWIR bands. 

1. Make the use of harmonics optional 
2. Make it easy to edit in the script the amount of harmonics to generate, which bands to generate them on, and the length and location of the time period within the year (i.e. Jan-May SWIR harmonic, June-Aug NDVI harmonic, etc).

**Comment by kyle-woodward** [comment]
Currently, I am thinking that exposing the necessary parameters to control this at the CLI with parsed user arguments would get unruly. I am thinking instead to create a new configuration file (maybe `features.py`), where the user will be able to edit both the list of indices they'd like harmonic bands generated for, as well as the intra-year time period (month-day) to do so. 

The structure could look something like this as a dictionary:
```
{
'NDVI': {'start': '01-01', 'end': '06-01'},
'EVI': {'start': '01-01', 'end': '12-31'}
}
```
The dictionary items will then be parsed by a helper function before it is passed to the modularized harmonics function

Steps:
* fully modularize the harmonics functionality, ensuring it can take a list of indices and start/end date strings
* build out dictionary -> harmonics band stacking flow, may require a few additional helper functions

The key will be not to further engineer the actual harmonics function to be specific to this user-control feature integration, but to build the user-control functionality around the harmonics function as it already behaves.

**Comment by kyle-woodward** [comment]
closed with PR #12
